R/tag_query.R

```diff
@@ -0,0 +1,1752 @@
+    # remove parent / child relationship
+    el$parent <- NULL
+    elParent$children[[childPos]] <- NULL
+  })
+}
+# Add siblings after each el
+tagQuerySiblingAfter <- function(els, ...) {
+  tagQueryMatchChildRev(els, function(elParent, el, childPos) {
+    tagInsertChildren(elParent, after = childPos, ...)
+  })
+}
+# Add siblings before each el
+tagQuerySiblingBefore <- function(els, ...) {
+  tagQueryMatchChildRev(els, function(elParent, el, childPos) {
+    tagInsertChildren(elParent, after = childPos - 1, ...)
+  })
+}
+# Replace all `el` objects with `...`
+tagQuerySiblingReplaceWith <- function(els, ...) {
+  tagQueryMatchChildRev(els, function(elParent, el, childPos) {
+    # Remove the current element
+    el$parent <- NULL
+    elParent$children[[childPos]] <- NULL
+    # Replace with ... content where the child was
+    tagInsertChildren(elParent, after = childPos - 1, ...)
+  })
+}
+
+
+tagQuerySetChildren <- function(els, ...) {
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    tagSetChildren(el, ...)
+  })
+}
+tagQueryChildrenEmpty <- function(els) {
+  # do not include any arguments.
+  # `dots_list()` returns an empty named list()
+  tagQuerySetChildren(els)
+}
+tagQueryChildrenAppend <- function(els, ...) {
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    tagInsertChildren(el, after = length(el$children), ...)
+  })
+}
+tagQueryChildrenPrepend <- function(els, ...) {
+  tagQueryChildrenInsert(els, after = 0, ...)
+}
+tagQueryChildrenInsert <- function(els, after, ...) {
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    tagInsertChildren(el, after = after, ...)
+  })
+}
+
+
+# Add attribute values
+tagQueryAttrsAdd <- function(els, ...) {
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    el <- tagAppendAttributes(el, ...)
+    el$attribs <- flattenTagAttribs(el$attribs)
+  })
+}
+# Remove attribute values
+tagQueryAttrsRemove <- function(els, attrs) {
+  attrs <- unlist(list2(attrs))
+  if (length(attrs) < 1) return()
+  if (!is.character(attrs)) {
+    stop("`attrs` must be a charcter vector of attributes to remove")
+  }
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    for (attrVal in attrs) {
+      el$attribs[[attrVal]] <- NULL
+    }
+  })
+}
+# Remove attribute values
+tagQueryAttrsEmpty <- function(els) {
+  tagQueryWalk(els, function(el) {
+    if (!isTagEnv(el)) return()
+    el$attribs <- list()
+  })
+}
+# Check if els have attributes
+tagQueryAttrHas <- function(els, attr) {
+  attr <- unlist(list2(attr))[[1]]
+  if (length(attr) != 1 || !is.character(attr)) {
+    stop("`attr` must be a single character value")
+  }
+  unlist(
+    tagQueryLapply(els, function(el) {
+      if (!isTagEnv(el)) return(FALSE)
+      tagHasAttribute(el, attr)
+    }),
+    use.names = FALSE
+  )
+}
+
+
+getCssClass <- function(class) {
+  class <- unlist(list2(class))
```

Comment:
```suggestion
  class <- as.character(list2(class))
```

Here and elsewhere that `unlist()` is used.